aiida_mlip/data/model.py

```diff
@@ -0,0 +1,220 @@
+        ----------------
+        kwargs : Any
+            Additional keyword arguments.
+        """
+        super().__init__(file, filename, **kwargs)
+        self.base.attributes.set("architecture", architecture)
+
+    def set_file(
+        self,
+        file: Union[str, Path],
+        filename: Optional[str] = None,
+        architecture: Optional[str] = None,
+        **kwargs: Any,
+    ) -> None:
+        """Set the file for the node.
+
+        Parameters
+        ----------
+        file : Union[str, Path]
+            Absolute path to the file.
+        architecture : [str]
+            Architecture of the mlip model.
+        filename : Optional[str], optional
+            Name to be used for the file (defaults to the name of provided file).
+
+
+        Other Parameters
+        ----------------
+        kwargs : Any
+            Additional keyword arguments.
+        """
+        super().set_file(file, filename, **kwargs)
+        self.base.attributes.set("architecture", architecture)
+
+    @classmethod
+    def local_file(
+        cls,
+        file: Union[str, Path],
+        architecture: str,
+        filename: Optional[str] = None,
+    ):
+        """Create a ModelData instance from a local file.
+
+        Parameters
+        ----------
+        file : Union[str, Path]
+            Path to the file.
+        architecture : [str]
+            Architecture of the mlip model.
+        filename : Optional[str], optional
+            Name to be used for the file (defaults to the name of provided file).
+
+        Returns
+        -------
+        ModelData
+            A ModelData instance.
+        """
+        file_path = Path(file).resolve()
+        return cls(file=file_path, architecture=architecture, filename=filename)
+
+    @classmethod
+    # pylint: disable=too-many-arguments
+    def download(
+        cls,
+        url: str,
+        architecture: str,
+        filename: Optional[str] = None,
+        cache_dir: Optional[Union[str, Path]] = None,
+        force_download: Optional[bool] = False,
+    ):
+        """Download a file from a URL and save it as ModelData.
+
+        Parameters
+        ----------
+        url : str
+            URL of the file to download.
+        architecture : [str]
+            Architecture of the mlip model.
+        filename : Optional[str], optional
+            Name to be used for the file (defaults to the name of provided file).
+        cache_dir : Optional[Union[str, Path]], optional
+            Path to the folder where the file has to be saved (defaults to "~/.cache/mlips/").
+        force_download : Optional[bool], optional
+            True to keep the downloaded model even if there are duplicates (default: False).
+
+        Returns
+        -------
+        ModelData
+            A ModelData instance.
+        """
+        cache_dir = Path(cache_dir if cache_dir else "~/.cache/mlips/")
+        arch_dir = cache_dir / architecture if architecture else cache_dir
```

Comment:
While I know the result due to precedence, at first glance this seems like it might create `cache_dir / cache_dir`
```suggestion
        arch_dir = (cache_dir / architecture) if architecture else cache_dir
```